src/components/checkbox/checkbox.template.tsx

```diff
@@ -4,6 +4,7 @@ interface CheckboxTemplateProps {
     disabled?: boolean;
```

Comment:
Replying here so comments can be threaded:

> Sorry about pinging for reviews. Converted to draft now because I noticed that the div is lacking the equivalent css class we have for disabled/required + it's not mutable after render (can't toggle a rendered limel-checkbox by `querySelector('limel-checkbox').indeterminate = true`).

No worries! 🙂

You _can_ use `querySelector('limel-checkbox').indeterminate = true` to change the state of the checkbox. I just did in the docs for this PR. My guess is that you're not finding the checkbox because `document.querySelector('limel-checkbox')` returns `null`, because `querySelector` doesn't penetrate shadow-roots. _But_, I'm having trouble seeing when this might be a legit way to control state? Surely this state should be held somewhere it can be controlled without finding the checkbox through the DOM?

Reply:
@adrianschmidt I was testing this as well when you wrote this comment and I can also properly update the state by changing the example to use `@Element` and `this.element.shadowRoot.querySelector('limel-checkbox').indeterminate`. I didn't actually come across that error in this branch but when using a limel-checkbox for the "select all" checkbox inside tabulator. I know we had some similar issues when manipulating Lime Elements components from FullCalendar (read DOM manipulation and not stencil re-render after state/prop changes), so I thought it was the same thing going here. I'll need to investigate my changes to limel-table where the checkbox won't update properly. Thanks for testing and confirming it's working!

Reply:
But wait, why are you updating the state of a component this way in the first place? There really seems to be something wrong if you need to perform DOM manipulation rather than setting the value of a `@Prop`? Can you send me a link to the code you're talking about? (The table stuff preferably, I'll try to keep out of FullCalendar for now, if possible 😅)

Reply:
I'm not sure if this is the case with my callback from Tabulator, I think it's something else, so never mind for now.  :)
In the table column header I need to return a DOM element, so I use `document.createElement('limel-checkbox')` to add the select all checkbox.

However, for the sake of FullCalendar it has no Stencil connector and uses preact internally so it's more similar to the case in the limel-table where we want to avoid re-render and toggle CSS instead (the table loader element for instance). 

What's important here is my concern about setting props manually (https://stenciljs.com/docs/javascript#setting-the-prop-manually), which both you and I have confirmed working.